File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/MallpayRefundRequest.cs
using System.Text;
using CsobGatewayClientExample.Communication.DataObjects.Ext;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects;

public class MallpayRefundRequest : SignBaseRequest
{
    [JsonProperty("extension")] public List<Extension>? Extensions;


    [JsonProperty("payId")] public string? PayId { get; set; }

    [JsonProperty("amount")] public long? Amount { get; set; }

    [JsonProperty("refundedItems", Required = Required.Always)]
    public List<MallpayOrderItemRef>? RefundedItems { get; set; }


    public override string ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, MerchantId);
        Add(sb, PayId);
        Add(sb, Dttm);
        Add(sb, Amount);
        if (RefundedItems != null)
            foreach (var item in RefundedItems)
                Add(sb, item.ToSign());
        DeleteLast(sb);
        return sb.ToString();
    }
}

File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/MallpayRefundRequest.java
package cz.monetplus.mips.eapi.v19.connector.entity;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.*;

import java.util.List;

@EqualsAndHashCode(callSuper = false)
@Data
@RequiredArgsConstructor
@JsonInclude(value = JsonInclude.Include.NON_NULL)
public class MallpayRefundRequest extends SignBase {
    @NonNull
    private String merchantId;
    @NonNull
    private String payId;
    @NonNull
    private Long amount;
    @NonNull
    private List<MallpayOrderItemRef> refundedItems;

    @Override
    public String toSign() {
        StringBuilder sb = new StringBuilder();
        add(sb, merchantId);
        add(sb, payId);
        add(sb, dttm);
        add(sb, amount);
        for(MallpayOrderItemRef item: refundedItems) add(sb, item.toSign());
        deleteLast(sb);
        return sb.toString();
    }
}
